src/drivers.rs

```diff
@@ -0,0 +1,73 @@
+
+use std::time::Duration;
+use std::path::{Path, PathBuf};
+
+use log::{debug, error};
+
+use crates_io_api::AsyncClient;
+
+use crate::PkgFmt;
+use crate::helpers::*;
+
+/// Fetch a crate by name and version from crates.io
+pub async fn fetch_crate_cratesio(name: &str, version: Option<&str>, temp_dir: &Path) -> Result<PathBuf, anyhow::Error> {
+    // Build crates.io api client and fetch info
+    let api_client = AsyncClient::new("cargo-binstall (https://github.com/ryankurte/cargo-binstall)", Duration::from_millis(100))?;
+
+    debug!("Fetching information for crate: '{}'", name);
+
+    // Fetch overall crate info
+    let info = match api_client.get_crate(name.as_ref()).await {
+        Ok(i) => i,
+        Err(e) => {
+            error!("Error fetching information for crate {}: {}", name, e);
+            return Err(e.into())
+        }
+    };
+
+    // Use specified or latest version
+    let version_num = match version {
+        Some(v) => v.to_string(),
+        None => info.crate_data.max_version,
+    };
+
+    // Fetch crates.io information for the specified version
+    // Note it is not viable to use a semver match here as crates.io
+    // appears to elide alpha and yanked versions in the generic response...
```

Comment:
so this probably needs to be _consistent_ with cargo, in which case maybe pulling the whole index would work better...

https://github.com/frewsxcv/rust-crates-index